### ISSUE TITLE
Error: Provided indices are out-of-bounds w.r.t. dense side with broadcasted shape

### ISSUE THREAD

**Issue by qianjiangcn** [NifTK/NiftyNet#272]
Before filing your issue, please note that it might be more appropriate for Stack Overflow, using the `niftynet` tag:
https://stackoverflow.com/questions/tagged/niftynet

Read the following to decide whether to file an issue here or ask a question on Stack Overflow:

I have been running into the error trying to use niftynet highRes2Dnet to perform 3D segmentation:

InvalidArgumentError (see above for traceback): Provided indices are out-of-bounds w.r.t. dense side with broadcasted shape
[[{{node worker_0/loss_function/map/while/mul}} = SparseDenseCwiseMul[T=DT_FLOAT, _class=["loc:@worke...tackPushV2"], _device="/job:localhost/replica:0/task:0/device:CPU:0"](worker_0/loss_function/map/while/SparseReshape, worker_0/loss_function/map/while/ones_like, worker_0/loss_function/map/while/SparseReshape:1, worker_0/loss_function/map/while/Softmax)]]

As one issue stated, this could happen when the largest voxel value in the discrete segmentation maps is greater than num_classes in the config file.
what do you mean largest voxel value in the discrete segmentation maps?
my spatial window is 32 and the num_classes in the config is 160.
Here is the config info and the size of input data is 128*256*256

[T1]
path_to_search = ../Training/Training_E2E
filename_contains = T1
filename_not_contains =
spatial_window_size = (32, 32, 32)
interp_order = 0
pixdim = (1.0, 1.0, 1.0)
axcodes = (R, A, S)

[parcellation]
path_to_search = ../Training/Training_E2E
filename_contains = Label
filename_not_contains =
spatial_window_size = (32, 32, 32)
interp_order = 0
pixdim = (1.0, 1.0, 1.0)
axcodes = (R, A, S)

############################## system configuration sections
[SYSTEM]
cuda_devices = ""
num_threads = 2
num_gpus = 1
model_dir = ./models/model_highres3dnet

[NETWORK]
name = highres3dnet
batch_size = 1
activation_function = prelu
volume_padding_size = 10

whitening = False
normalisation = True
normalise_foreground_only=True
foreground_type = mean_plus
histogram_ref_file = ./example_volumes/monomodal_parcellation/standardisation_models.txt
cutoff = (0.001, 0.999)

[TRAINING]
sample_per_volume = 32
rotation_angle = (-10.0, 10.0)
scaling_percentage = (-10.0, 10.0)
lr = 0.0001
loss_type = Dice
starting_iter = 0
save_every_n = 5
max_iter = 100
max_checkpoints = 10

############################ custom configuration sections
[SEGMENTATION]
image = T1
label = parcellation
output_prob = False
num_classes = 32
label_normalisation = True

Thank you very much !
